pkg/controllers/managementuser/resourcequota/resource_quota_sync.go

```diff
@@ -516,3 +468,21 @@ func completeLimit(existingLimit *v32.ContainerResourceLimit, defaultLimit *v32.
 	}
 	return &newLimit, nil
 }
+
+// zeroOutResourceQuotaLimit takes a resource quota limit and a list of resources exceeding the quota,
+// and returns a new quota limit with exceeded resources zeroed out.
+func zeroOutResourceQuotaLimit(limit *v32.ResourceQuotaLimit, exceeded corev1.ResourceList) (*v32.ResourceQuotaLimit, error) {
+	limitMap, err := convert.EncodeToMap(limit)
+	if err != nil {
+		return nil, err
+	}
+
+	for k := range exceeded {
+		resource := string(k)
+		limitMap[resource] = "0"
+	}
+
+	toReturn := &v32.ResourceQuotaLimit{}
+	err = convert.ToObj(limitMap, toReturn)
```

Comment:
I think that we do something similar to this (convert from a map to an object) in `completeQuota`. There, we use:

```
mapstructure.Decode(newLimitMap, &toReturn)
```
Would you mind elaborating on why you used `convert.ToObj` here rather than `mapstructure.Decode`? I don't have a preference, but I would think that we would want to use the same library in each case here.

Reply:
I used `convert.ToObj` because it's used in the existing `convertResourceListToLimit` function. There should not be a difference in our use-case, but I can switch to `mapstructure.Decode`.

Reply:
But then we might as well be not using `convert.EncodeToMap`, since`mapstructure.Decode` can also do this, if I understand correctly...

Reply:
I'm not particularly partial to either method - it just seems like they are doing the same thing and I would rather have us only depend on one library rather than two.

Reply:
I will only use our `convert` package. Looks like `mapstructure` needs a bit more configuration, because it changes the case of the key names in maps, and this breaks things in at least one case. I verified that using our `convert` does not change expected behavior.